src/SpectatorView.Native/SpectatorView.Compositor/Compositor/AzureKinectFrameProvider.h

```diff
@@ -23,65 +20,41 @@ class AzureKinectFrameProvider : public IFrameProvider
     virtual LONGLONG GetTimestamp(int frame) override;
     virtual LONGLONG GetDurationHNS() override;
     virtual void Update(int compositeFrameIndex) override;
-    virtual ProviderType GetProviderType() override;
+    virtual ProviderType GetProviderType() override { return _providerType; }
     virtual bool IsEnabled() override;
     virtual bool SupportsOutput() override;
     virtual void Dispose() override;
     virtual bool OutputYUV() override;
 
     virtual int GetCaptureFrameIndex() override
     {
-        return _captureFrameIndex;
+        return cameraInput == nullptr ? 0 : cameraInput->GetCaptureFrameIndex();
     }
    
    virtual bool IsCameraCalibrationInformationAvailable() override
     {
         return true;
     }
 
-    virtual void GetCameraCalibrationInformation(CameraIntrinsics* calibration) override;
+   virtual void GetCameraCalibrationInformation(CameraIntrinsics* calibration) override { cameraInput->GetCameraCalibrationInformation(calibration); }
```

Comment:
>cameraInput [](start = 90, length = 11)

nit: not sure when this is called but we should probably make sure that this camera input isn't nullptr